### DISCUSSION TITLE
"Clip/Device View" command not working

### DISCUSSION THREAD

**Discussion by ll-dasein-ll** [KBLiveSolutions/open.control#19] [Troubleshooting] [answered]
Hi -

Just a quick one - I've just noticed that the "Clip/Device View" command doesn't seem to be working for me.  I've tried assigning it to a few different buttons and tested in Session and Arrangement view, but nothing happens when I press it.

I'm running the latest firmware and using the latest editor (that includes the Crossfader command)

Thank you!

**Comment by KBLiveSolutions** [comment]
Hey Martin

Thanks for your feedback, I'll have a look into this ASAP.
Cheers
p

**Reply by ll-dasein-ll** [reply]
Hey Pierre -

Did you have any joy with this problem?

Cheers,
Martin

**Comment by KBLiveSolutions** [comment, marked as answer]
Sorry for the delay.
I had time to look into this. I kinda fixed it, although, it's not perfect : the API doesn't tell whether it's a Clip or a Device that is currently displayed.
So sometimes it will take 1 press to toggle between the views, sometimes 2. I guess it's better than nothing.
You can check it by [downloading](https://github.com/KBLiveSolutions/open.control-Remote-Script/archive/refs/heads/main.zip) and installing the script again.

**Reply by ll-dasein-ll** [reply]
> the API doesn't tell whether it's a Clip or a Device that is currently displayed.

Well that's a bit annoying, isn't it!?  Thanks very much for the fix though, I'll test it out tonight - I'm sure it won't be difficult to get used to the little quirk. :-)

**Comment by ll-dasein-ll** [comment]
I've finally managed to test this out and it works a treat - I'm not finding that it needs 2 clicks at all.  However, the problem I have now is that it seems to have broken the Track Select function on my encoders again (I posted about this here a few weeks ago).  I've tried reverting back to earlier scripts and even back to firmware 1.4, but I can't get Track Select working at all anymore.

**Comment by ll-dasein-ll** [comment]
Hi Pierre -

I've updated the thread on Github to mention this, but since it's marked as
answered I'm not sure if you get notifications for it any more.  The
clip/device setting now works perfectly for me, but now that that's
working, the Track Select function has stopped working on my encoders
again!  Is there any chance you could take a look please?

Out of interest, is stuff like this simple enough for me to have a try at
fixing myself, or might I screw things up if I don't really know what I'm
doing with coding?  I'm not completely clueless and do stuff with DietPi
and Powershell and stuff, but I would almost definitely end up breaking
things quite regularly if I start messing with the code.  As long as I'm
confident that I could restore everything by reinstalling the firmware and
couldn't easily brick the controller, I'd gladly have a try myself rather
than bothering you with every little problem I encounter :-)

Cheers,
Martin

**Reply by KBLiveSolutions** [reply]
Hi Martin

Sorry for the lack of answer, I've been pretty busy.

Please download the Remote Script and install again, the Track Select (and also Crossfader) should work fine now.
I just forgot to paste those 2 lines in the previous fix.

Regarding trying to fix things yourself, you would need some Python knowledge and a good understanding of the Live API.
If I find some time, I would like to make some kind of tutorial to explain what's going on.
But it won't be soon I am afraid.

Cheers 
p

**Comment by ll-dasein-ll** [comment]
No probs Pierre, and thanks very much for the update - I've not tested it
yet but I'm assuming it'll all work again now :-)

I think I'll leave the coding alone for now but might have a look over the
summer or something.  It's something I'd like to dip into but it sounds
like I've got a lot of studying to do first!

Cheers,
Martin

**Comment by ll-dasein-ll** [comment]
Hey Pierre - 

Really sorry about this (and for taking weeks to update you - I've not had a chance to check this for a while!), but the Track Select function still isn't working for me on my encoders after downloading the latest remote script.  The encoders are definitely okay - all other actions I've tested work, just not the Track Select.  

Hoping it's easy to fix - I dipped into the code but it's way above my understanding so i thought I'd better ask for help here again :-)

Cheers!

**Reply by KBLiveSolutions** [reply]
Hello

I'm away for a few days but I should have some time to look into this by the end of the month, I'll keep you updated.
Cheers

**Reply by ll-dasein-ll** [reply]
no problem - thank you 😁

**Reply by ll-dasein-ll** [reply]
Hey Pierre -

I'm not sure what I've done differently, but I ended up having some strange glitches with the opencontrol the other day so downgraded the firmware and remote script, then upgraded them both again and the Track Select function is working again!  Not sure why it didn't work the last few times I tried doing this, but it's looking good now.  I've now got another really weird problem though - I'll start a new thread for this because it's completely unrelated to this.

Thanks again!
Martin
